File: bindings/python/examples/remote_embed.py
from __future__ import annotations

from _common import (
    CogentClient,
    add_openai_remote,
    print_embedding,
    read_remote_args,
)


def main() -> None:
    model, input_text = read_remote_args("CogentClient remote embedding smoke input.")
    client = CogentClient()
    endpoint = add_openai_remote(client, model)
    run = client.embed(
        input_text,
        endpoint=endpoint,
    )
    print_embedding(run.result())


if __name__ == "__main__":
    main()

File: bindings/node/examples/remote_embed.mjs
import {
  CogentClient,
  addOpenAiRemote,
  printEmbedding,
  readRemoteArgs,
} from './_common.mjs';

const { model, input } = readRemoteArgs('CogentClient remote embedding smoke input.');
const client = new CogentClient();
const endpoint = addOpenAiRemote(client, model);
const result = await client.embed({
  endpoint,
  input,
}).response;
printEmbedding(result);
